### DISCUSSION TITLE
Debounce revalidation

### DISCUSSION THREAD

**Discussion by Miloshinjo** [vercel/swr#1438] [Q&A]
Hi, I am a new SWR user and I hit a particular problem that I still didn't find in the docs, so I'll present it here.

I have a basic to-do, shopping-list app, where you can add items or remove them etc..

This is how I see it. 

User removes an item from the list, it is removed from the cache, DELETE req is called and revalidation kicks in with `mutate` function. 

In my case, I click to delete many items in succession, so several revalidation functions kick in. So sometimes it happens that "state" data is shown, after cache displays correct data because revalidation kicked in for some previous cases.

I also need to mention that I have `refreshInterval: 5000` set, so that may be also making the data "stale"

How does this work? Is there a way to "debounce" revalidation somehow, so that only last revalidation request "kicks-in" after fast clicking?

 I may have set up this wrongly and this works out of the box, but again I say, I only started using it, and I like it :)

**Comment by shuding** [comment]
> Is there a way to "debounce" revalidation somehow, so that only last revalidation request "kicks-in" after fast clicking?

This mechanism should be implemented inside SWR already — however if it doesn’t work like that it _might_ be a bug. If you notice that and are able to provide a reproduction, please file an issue! Thanks!
